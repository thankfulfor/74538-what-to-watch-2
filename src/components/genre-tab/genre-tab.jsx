import * as React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer.js';

export const GenreTab = (props) => {
  const {genres, onTabClick} = props;

  const tabClickHandler = (evt) => {
    evt.preventDefault();
    onTabClick(evt.target.textContent);
  };

  return (
    <ul className="catalog__genres-list">
      {genres.map((genre, i) => {
        return (
          <li className="catalog__genres-item" key={`${genre}-${i}`}>
            <a onClick={tabClickHandler} href="#" className="catalog__genres-link">{genre}</a>
          </li>
        );
      })}
    </ul>
  );
};

GenreTab.propTypes = {
  genres: PropTypes.array.isRequired,
  onTabClick: PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  genre: state.genre,
});

const mapDispatchToProps = (dispatch) => ({
  onTabClick: (genre) => {
    dispatch(ActionCreator.changeFilterByGenre(genre));
    dispatch(ActionCreator.getFilmListByGenre(genre));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(GenreTab);
