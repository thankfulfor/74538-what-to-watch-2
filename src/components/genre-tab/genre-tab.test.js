import React from 'react';
import renderer from 'react-test-renderer';
import {GenreTab} from './genre-tab.jsx';

const genres = [``];

it(`Is genre catalog tab rendered`, () => {
  const tabClickHandler = jest.fn();

  const tree = renderer
    .create(<GenreTab
      genres={genres}
      onTabClick={tabClickHandler}
    />);

  expect(tree).toMatchSnapshot();
});
