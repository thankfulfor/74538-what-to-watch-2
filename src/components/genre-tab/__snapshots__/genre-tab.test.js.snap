// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is genre catalog tab rendered 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      
    </a>
  </li>
</ul>
`;
